packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -180,6 +185,34 @@ class WidgetTester extends WidgetController implements HitTestDispatcher {
     return TestAsyncUtils.guard(() => binding.pump(duration, phase));
   }
 
+  /// Calls [pump] with the given `duration` until there are no longer any
+  /// transient callbacks scheduled.
+  ///
+  /// This essentially waits for all animations to have completed.
+  ///
+  /// This function will never return (and the test will hang and eventually
+  /// time out and fail) if there is an infinite animation in progress (for
+  /// example, if there is an indeterminate progress indicator spinning).
+  ///
+  /// In general, it is better practice to figure out exactly why each frame is
```

Comment:
I was very to happy to see this new function  and then a little disappointed to see that you're suggesting that using it indicates a lack of character :-).